libs/s25main/world/GameWorldViewer.cpp

```diff
@@ -111,6 +111,12 @@ Visibility GameWorldViewer::GetVisibility(const MapPoint pt) const
     if(GAMECLIENT.IsReplayModeOn() && GAMECLIENT.IsReplayFOWDisabled())
         return VIS_VISIBLE;
 
+    // Im Wirtschaftsmodus und Spiel vorbei? Dann auch alles sichtbar
+    if(GetWorld().econHandler && GetWorld().econHandler->globalVisibility())
```

Comment:
Hm, maybe a better way would be to make the world update all nodes fow members and set them to visible? Because this is a "set once" at the end of the econ mode, isn't it?

Reply:
I had tried this because I agree with your conceptual preference. But setting every node visible individually can take multiple seconds on very large maps, while the way I decided to use here takes no noticeable time. Also this same way is used for defeated players, which is also set at most once for each player. 

However, if you can give me a hint how to solve the performance issues with updating all nodes I am happy to change this.

Reply:
Multiple seconds? And you simply loop over the nodes and fow entries?
Hm... that's hard then. Sure that this is optimised (release) mode already? Maybe we need to restructure the array from AoS to SoA

Reply:
I just looked at it more thoroughly. Previously I had called `GameWorldGame::MakeVisible(const MapPoint pt, const unsigned char player)` which seemed the correct way. But I now think this is slow because it updates the minimap and potentially lua scripts etc. after every single point. 

So I added a function to world directly working on the nodes and then a function to the GameInterface to explicitly update the minimap and colors afterwards. This seems to be without noticeable performance issues. (Note that I only actually implemented this interface method in dskGameInterface, because that's the only point where it's necessary for now)